helpers/test-helper.js

```diff
@@ -435,6 +437,23 @@ export async function submitAndForwardTimeToDispute(clients, test) {
     await clients[i].submitRootHash();
   }
   await forwardTime(MINING_CYCLE_DURATION / 2, test);
+
+  // If there are multiple submissions, ensure they are different
+  if (clients.length > 1) {
+    for (let i = 1; i < clients.length; i += 1) {
```

Comment:
Mm good point, you'd need an inner loop to make all the pairwise comparisons.

Reply:
Do we want to allow that actually? That is, make this check so that at least 2 of the client submissions differ? I had a problem with this scenario where 3 clients submitted: 1 good and 2 bad but using the same hash, which necessitated [this change](https://github.com/JoinColony/colonyNetwork/pull/538/commits/4d2969590db00e7787072a24a435142cc21043c3#diff-de6296b0343869cb2b046fcc5e97c6e3R545)

Reply:
That's a good point also.

Reply:
Maybe just count the number of different submissions, and compare that number to a parameter passed in?

Reply:
I've pushed what I think is a much more graceful solution using promises and sets. We additionally ensure that this helper function is called with at least 2 clients, otherwise there's really no dispute. Only one test was using that wrongly and so I've moved it to use `advanceMiningCycleNoContest` which is the right function when there's one client and no dispute.

Reply:
I am not liking a dedicated parameter in the function to indicate additional logic checks like hoe many submissions should be different as it's complicating the helper unnecessarily and when using that in code I've personally found it more cumbersome to write. So for now I'd like to stick with a single function that cleanly checks all submissions are different. For other scenarios like in that one test we can do the submissions manually.

Reply:
Yeah, if we're happy with the idea that tests where some submissions are intended to be the same have to not use this function, then this is much better.